tools/jit/gen_jit_dispatch.py

```diff
@@ -58,64 +63,126 @@ def is_jit_op(decl):
+            else:
+                tensor_id = iter(count(start=1))
+                args = ['TensorTemporary(inputs[{}]).value()'.format(next(tensor_id))
+                        if is_tensor_arg(arg) else arg['name']
+                        for arg in arguments[1:]]
+                call = CALL_METHOD.substitute(name=name, args=args)
+
+            constructor = CONSTRUCTOR.substitute(descriptor=descriptor, name=name, call=call,
+                                                 kw_assignments=kw_assignments,
+                                                 pos_assignments=pos_assignments,
+                                                 # num_inputs is only used in AutogradClosure, which
+                                                 # is going to be removed soon anyway. There's no good value
+                                                 # we can provide for cat.
+                                                 num_inputs=num_inputs if num_inputs != "*" else 0)
+            if descriptor not in ops:
+                ops[descriptor] = (constructor, decl)
             else:
-                tensor_id = iter(count(start=0))
-                args = ['TensorTemporary(inputs[{}]).value()'.format(
-                    next(tensor_id)) if is_tensor_arg(arg) else arg['name']
-                    for arg in arguments]
-            call = CALL_NAMESPACE.substitute(name=name, args=args)
-        else:
-            tensor_id = iter(count(start=1))
-            args = ['TensorTemporary(inputs[{}]).value()'.format(next(tensor_id)) if is_tensor_arg(arg) else arg['name']
-                    for arg in arguments[1:]]
-            call = CALL_METHOD.substitute(name=name, args=args)
-
-        constructor = CONSTRUCTOR.substitute(descriptor=descriptor, name=name, call=call,
-                                             assignments=assignments,
-                                             # num_inputs is only used in AutogradClosure, which
-                                             # is going to be removed soon anyway. There's no good value
-                                             # we can provide for cat.
-                                             num_inputs=num_inputs if num_inputs != "*" else 0)
-        assert descriptor not in ops, descriptor
-        ops[descriptor] = constructor
+                # There are still some situations where we can recover from an ambiguous descriptor.
+                def fail():
+                    raise RuntimeError('descriptor conflict: ' + descriptor)
+
+                # If there are two overloads with the same descriptor, that differ only by a type of a
+                # single argument, where one of them takes a tensor, while another one takes an
+                # at::Scalar as a positional scalar arg, then prefer the tensor overload.
+                # It should get broadcasted correctly.
+                if descriptor in prefer_tensor_overload:
```

Comment:
Is it possible to just have a list of exclusions instead? This recover code is pretty long. e.g. just have `[ ('lt-2', 1), ... ` and then have a check if the descriptor is `lt-2` and arg_type[1] == 'Scalar', then don't bother adding it.